src/Node/TableNode.php

```diff
@@ -362,6 +362,14 @@ public function mergeRowsFromTable(TableNode $node)
         }
     }
 
+    /**
+     * Returns true if table is empty (has no rows, but it might have a header).
+     */
+    public function isEmpty(): bool
+    {
+        return count($this->table) <= 1;
+    }
+
```

Comment:
Is this method required? I don't see it used anywhere.

I think it could cause confusion - for a table attached to a step node it's entirely valid to have a 1-row table which is not considered "empty" e.g.

```gherkin
When I fill in the following:
  | username | acoulton |
```

It's only `Examples` tables that are expected to have a header and subsequent data rows?

Reply:
Not anymore 👍